internal/store/assessment.go

```diff
@@ -65,32 +66,60 @@ func (a *AssessmentStore) Get(ctx context.Context, id uuid.UUID) (*model.Assessm
 	return &assessment, nil
 }
 
-func (a *AssessmentStore) Create(ctx context.Context, assessment model.Assessment, inventory api.Inventory) (*model.Assessment, error) {
-	// Create the assessment first
-	result := a.getDB(ctx).Clauses(clause.Returning{}).Create(&assessment)
-	if result.Error != nil {
-		if errors.Is(result.Error, gorm.ErrDuplicatedKey) {
-			return nil, ErrDuplicateKey
+// Create creates an assessment with a snapshot
+// If inventory is nil, creates a pending snapshot for async processing
+// If inventory is provided, creates a ready snapshot with the inventory
+func (a *AssessmentStore) Create(ctx context.Context, assessment model.Assessment, inventory *api.Inventory) (*model.Assessment, error) {
+	db := a.getDB(ctx)
+
+	// Wrap both creates in a transaction to prevent dangling assessment
+	err := db.Transaction(func(tx *gorm.DB) error {
```

Comment:
"db" handler should be transaction here already started by the service. No need to begin a new transaction.

Reply:
as general remark: the store layer does not know about transactions which are started and committed at service layer.